apps/desktop-wallet/src/features/thumbnails/NFTThumbnail.tsx

```diff
@@ -38,71 +38,77 @@ interface NFTThumbnailProps {
-        }}
-        onClick={onClick}
-      >
-        {!isLoading ? (
-          src ? (
-            nft.dataType === 'image' ? (
-              <Image src={src} height={size} width={size} />
-            ) : nft.dataType === 'video' ? (
-              autoPlay ? (
-                <video src={src} autoPlay={autoPlay} loop width="100%" height="100%" preload="auto" muted playsInline />
-              ) : (
-                <VideoThumbnail videoUrl={src} showPlayIcon={showPlayIconIfVideo} playOnHover={playOnHover} />
-              )
-            ) : (
-              <ErrorMessage>{t('Unsupported media type')}</ErrorMessage>
-            )
-          ) : (
-            <ErrorMessage>{t('No media')}</ErrorMessage>
-          )
-        ) : (
-          <SpinnerContainer>
-            <Spinner />
-          </SpinnerContainer>
-        )}
-      </Container>
+const NFTThumbnail = ({ border, borderRadius, onClick, hideIfError, ...props }: NFTThumbnailProps) => {
+  const { error } = useFetchNft({ id: props.nftId })
+
+  if (hideIfError && error) return null
+
+  return (
+    <NftThumbnailStyled border={border} borderRadius={borderRadius} onClick={onClick} size={props.size}>
+      <NftThumbnailMedia {...props} />
+    </NftThumbnailStyled>
+  )
+}
+
+export default NFTThumbnail
+
+const NftThumbnailMedia = memo(({ nftId, size, ...props }: NFTThumbnailProps) => {
+  const { t } = useTranslation()
+
+  const { data: nft, isLoading, error: fetchNftError } = useFetchNft({ id: nftId })
+
+  if (isLoading) return <SkeletonLoaderStyled height={size} />
+
+  if (fetchNftError || !nft || (nft && !nft.image)) return <MissingMediaPlaceholder />
+
+  if (nft.dataType === 'image') return <NftThumbnailMediaImage src={nft.image} alt={nft.description} size={size} />
+
+  if (nft.dataType === 'video') return <NftThumbnailMediaVideo {...props} src={nft.image} />
+
+  return <ErrorMessage>{t('Unsupported media type')}</ErrorMessage>
+})
+
+interface NftThumbnailMediaImageProps {
+  src: string
+  alt?: string
+  size?: string
+}
+
+const NftThumbnailMediaImage = memo(({ src, alt, size }: NftThumbnailMediaImageProps) => {
+  const [error, setError] = useState(false)
+
+  if (error) return <MissingMediaPlaceholder />
+
+  return <Image src={src} alt={alt} height={size} width={size} onError={() => setError(true)} />
+})
+
+interface NftThumbnailMediaVideoProps
+  extends Pick<NFTThumbnailProps, 'playOnHover' | 'autoPlay' | 'showPlayIconIfVideo'> {
+  src: string
+}
+
+const NftThumbnailMediaVideo = memo(
+  ({ src, showPlayIconIfVideo, playOnHover, autoPlay }: NftThumbnailMediaVideoProps) =>
+    autoPlay ? (
+      <video src={src} autoPlay={autoPlay} loop width="100%" height="100%" preload="auto" muted playsInline />
+    ) : (
+      <VideoThumbnail videoUrl={src} showPlayIcon={showPlayIconIfVideo} playOnHover={playOnHover} />
     )
-  }
 )
 
-const Container = styled.div`
+const MissingMediaPlaceholder = () => (
+  <ErrorMessage>
+    <CameraOff opacity={0.8} />
```

Comment:
I replaced the "X" with the same icon as the one we used to have:

<img width="203" alt="image" src="https://github.com/user-attachments/assets/a57ed419-e9d6-4a8b-bb98-e0d4def89d14">